doc/source/whatsnew/v0.21.0.txt

```diff
@@ -60,6 +60,7 @@ Other API Changes
 ^^^^^^^^^^^^^^^^^
 
 - Moved definition of ``MergeError`` to the ``pandas.errors`` module.
+- :func:`eval` will now raise a ``ValueError`` in cases where we cannot operate properly on the ``target`` parameter, or when an inplace operation is specified but there is no item assignment in the expression (:issue:`16732`)
```

Comment:
I think it makes sense to have a couple of examples in a sub-section on what is happening. Also feel free to expand the Examples section of the doc-string.